users/vieting/experiments/switchboard/ctc/feat/experiments.py

```diff
@@ -628,3 +629,138 @@ def run_mel_baseline():
         os.path.join(gs.ALIAS_AND_OUTPUT_SUBDIR, "report_mel_baseline.csv"),
         values=report.get_values(),
         template=report.get_template())
+
+def run_mel_audio_perturbation():
+    gs.ALIAS_AND_OUTPUT_SUBDIR = "experiments/switchboard/ctc/feat/"
+
+    (
+        returnn_datasets, rasr_loss_corpus_path, rasr_loss_corpus_segments, rasr_loss_lexicon_path, dev_corpora
+    ) = get_datasets(preprocessing=True)
+    returnn_args = {
+        "batch_size": 10000,
+        "rasr_binary_path": RASR_BINARY_PATH,
+        "rasr_loss_corpus_path": rasr_loss_corpus_path,
+        "rasr_loss_corpus_segments": rasr_loss_corpus_segments,
+        "rasr_loss_lexicon_path": rasr_loss_lexicon_path,
+        "datasets": returnn_datasets,
+    }
+    feature_args = {
+        "class": "LogMelNetwork",
+        "wave_norm": True,
+        "frame_size": 200,
+        "frame_shift": 80,
+        "fft_size": 256
+    }
+
+    nn_args, report_args_collection = get_nn_args_baseline(
+        nn_base_args={
+            "lgm80_conf-wei-oldspecaug-audio_perturbation": dict(
+                returnn_args={
+                    "conformer_type": "wei",
+                    "specaug_old": {},
+                    "audio_perturbation": True,
+                    "extra_args": {
+                        "audio_perturb_args": {
+                            "speed": {"prob": 0.6, "minimum": 0.88, "maximum": 1.12},
+                            "tempo": {"prob": 0.6, "minimum": 0.83, "maximum": 1.17},
+                        },
+                        "audio_perturb_runner": CodeWrapper("WaveformPerturbation(**audio_perturb_args)")
+                    },
+                    **returnn_args
+                },
+                feature_args=feature_args,
+                lr_args={
+                    "peak_lr": 4e-4, "start_lr": 1.325e-05, "end_lr": 1e-5,
+                    "increase_epochs": 119, "peak_epochs": 2, "decrease_epochs": 119, "final_epochs": 0,
+                },
+                report_args={
+                    "architecture": "conf-wei", "lr": "wei_peak_4e-4", "speed": "0.6_0.88_1.12",
+                    "tempo": "0.6_0.83_1.17",
+                },
+            ),
+        },
+        num_epochs=450,
+        prefix="conformer_bs10k_"
+    )
+
+    returnn_configs = {}
```

Comment:
Starting from here, there is a large code duplication with `run_mel_baseline()`. You could add a function `run_mel_nn_args()` which gets `nn_args` and maybe `report_name` and then does the rest here. That could be used by `run_mel_baseline()` and `run_mel_audio_perturbation()`.

Reply:
Done